connection.go

```diff
@@ -303,6 +308,29 @@ func (sc *snowflakeConn) ExecContext(ctx context.Context, query string, args []d
 }
 
 func (sc *snowflakeConn) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Rows, error) {
+	qid, _ := getResumeQueryID(ctx)
+	if qid == "" {
+		return sc.queryContextInternal(ctx, query, args)
+	}
+
+	// so there is a queryID in context for which we want to fetch the result
+	if !queryIDRegexp.MatchString(qid) {
```

Comment:
`getResumeQueryID` can return a `err`, right now, the `err` is ignored, but we can move this UUID regular expression checking into the `getResumeQueryID` and return the err there.